gtsam/hybrid/HybridBayesNet.cpp

```diff
@@ -14,3 +14,96 @@
  */
 
 #include <gtsam/hybrid/HybridBayesNet.h>
+
+namespace gtsam {
+
+/* ************************************************************************* */
+/// Return the DiscreteKey vector as a set.
+static std::set<DiscreteKey> DiscreteKeysAsSet(const DiscreteKeys &dkeys) {
+  std::set<DiscreteKey> s;
+  s.insert(dkeys.begin(), dkeys.end());
+  return s;
+}
+
+/* ************************************************************************* */
+HybridBayesNet HybridBayesNet::prune(
+    const DecisionTreeFactor::shared_ptr &discreteFactor) const {
```

Comment:
nit: I would argue that we should only allow pruning through a GaussianMixture

Reply:
I'm not sure I follow. Isn't pruning done on the discrete decision tree and then reflected on the full hybrid tree?

A GaussianMixture would only have the tree on it's keys right? Not the full graph